File: src/main/java/me/fzzyhmstrs/amethyst_core/interfaces/DamageTracking.java
package me.fzzyhmstrs.amethyst_core.interfaces;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.damage.DamageSource;
import net.minecraft.item.ItemStack;
import org.jetbrains.annotations.Nullable;

public interface DamageTracking {

    default void onWearerDamaged(ItemStack stack, LivingEntity wearer, @Nullable LivingEntity attacker, DamageSource source, Float amount){

    }
}

File: src/main/kotlin/me/fzzyhmstrs/amethyst_core/item_util/interfaces/DamageTracking.kt
package me.fzzyhmstrs.amethyst_core.item_util.interfaces

import net.minecraft.entity.LivingEntity
import net.minecraft.entity.damage.DamageSource
import net.minecraft.item.ItemStack

interface DamageTracking {
    fun onWearerDamaged(stack: ItemStack, wearer: LivingEntity, attacker: LivingEntity?, source: DamageSource, amount: Float)
}